index.js

```diff
@@ -39,29 +44,50 @@ function decode(bmsg) {
   return result
 }
 
-// input: json encoded msg from db
-function encode(msg) {
+/**
+ * Encode a database "msg value" to a bendy-butt msg in a Buffer.
+ *
+ * @param {Object} msgVal an object compatible with ssb/classic `msg.value`
+ * @returns {Buffer} a bendy-butt message encoded with `bencode`
+ */
+function encode(msgVal) {
   const contentSection =
-    typeof msg.content === 'string' && msg.content.endsWith('.box2')
-      ? bfe.encodeBendyButt(msg.content)
+    typeof msgVal.content === 'string' && msgVal.content.endsWith('.box2')
+      ? bfe.encodeBendyButt(msgVal.content)
       : [
-          bfe.encodeBendyButt(msg.content),
-          bfe.encodeBendyButt(msg.contentSignature),
+          bfe.encodeBendyButt(msgVal.content),
+          bfe.encodeBendyButt(msgVal.contentSignature),
         ]
 
   return bencode.encode([
     [
-      bfe.encodeBendyButt(msg.author),
-      bfe.encodeBendyButt(msg.sequence),
-      bfe.encodeBendyButt(msg.previous),
-      bfe.encodeBendyButt(msg.timestamp),
+      bfe.encodeBendyButt(msgVal.author),
+      bfe.encodeBendyButt(msgVal.sequence),
+      bfe.encodeBendyButt(msgVal.previous),
+      bfe.encodeBendyButt(msgVal.timestamp),
       contentSection,
     ],
-    bfe.encodeBendyButt(msg.signature),
+    bfe.encodeBendyButt(msgVal.signature),
   ])
 }
 
-// returns a classic compatible json object
+/**
+ * Creates a new bendy-butt message that can be appended to a bendy-butt feed.
+ *
+ * FIXME: this method does not seem to belong to `ssb-bendy-butt` because it is
```

Comment:
I was also wondering about this. I have been holding-off on implementing this in Rust for the moment. I would tend towards having `ssb-meta-feeds` import the necessary encoders (BFE and Bendy) and performing the object creation itself.

Reply:
I'm working on something to address this. I'm repurposing `create` to `encodeNew` and I'm abstracting away some details, and I intend to update the [bendy-butt-spec](https://github.com/ssb-ngi-pointer/bendy-butt-spec) to not mention anything metafeeds or subfeeds related.

Reply:
> Technically people could use bendy-butt for purposes other than metafeeds, right?

Absolutely. Decouple the tools so they can be useful in many contexts.